### PR TITLE
Select the fetcher based on the environment only once

### PR DESCRIPTION
**Pull request by NSeydoux**
The appropriate fetch method is selected when constructing the Fetcher object, instead of running the check at each query.

- [X] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: src/util/Fetcher.ts [hunk 1]
```diff
@@ -30,11 +30,18 @@ export interface IFetcher {
 }
 
 export default class Fetcher implements IFetcher {
+  private fetcher: typeof _fetch;
+
+  constructor() {
+    if (typeof window !== "undefined" && typeof window.fetch === "function") {
```
Comment (RubenVerborgh) [inline src/util/Fetcher.ts:36]: Or even above the level of `Fetcher` itself, but I take it there will not be a lot of instantiations?

File: src/util/Fetcher.ts [hunk 2]
```diff
@@ -30,11 +30,18 @@ export interface IFetcher {
...
+      this.fetcher = window.fetch;
+    } else {
+      this.fetcher = _fetch;
+    }
+  }
+
   async fetch(url: RequestInfo | URL, init?: RequestInit): Promise<Response> {
```
Comment (RubenVerborgh) [inline src/util/Fetcher.ts:43]: I still find it slightly weird that we decide here to offer other parameters than the regular `fetch` method; I don't think it's our task to also accept `URL`. From a drop-in replacement, you'd expect that it works no differently from the original.

Comment (pmcb55) [reply src/util/Fetcher.ts:43]: Yeah - I'm no expert in this area, but I agree the method signature should match standard fetch() **_exactly_** (if possible): https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch. Is there a specific reason to diverge?

**Comment by NSeydoux** [comment]
Due to the architectural changes leading the lib not to be isomorphic anymore, this is no longer relevant.

**Pull request closed without merging**